zerver/actions/scheduled_messages.py

```diff
@@ -257,8 +262,17 @@ def send_scheduled_message(scheduled_message: ScheduledMessage) -> None:
     assert not scheduled_message.delivered
     assert not scheduled_message.failed
 
-    # It's currently not possible to use the reminder feature.
-    assert scheduled_message.delivery_type == ScheduledMessage.SEND_LATER
+    if scheduled_message.delivery_type == ScheduledMessage.REMIND:
+        # Reminder messages are always sent from the notification bot.
+        message_id = internal_send_private_message(
+            get_system_bot(settings.NOTIFICATION_BOT, scheduled_message.realm.id),
+            get_user_profile_by_id(scheduled_message.recipient.type_id),
+            scheduled_message.content,
+        )
+        scheduled_message.delivered_message_id = message_id
+        scheduled_message.delivered = True
+        scheduled_message.save(update_fields=["delivered", "delivered_message_id"])
+        return
```

Comment:
So I think one of the major product questions about this implementation that we should discuss is whether we like the design that reminders construct the message content for the reminder at the time the reminder is scheduled (as this draft PR does) or whether we want to construct it at the time that the reminder is meant to be sent.

(This will affect things like whether the link points to the current location of the message if it's been moved to a different stream/topic).

If we want to construct the message content at the time that was scheduled, that may point to making a new set of database tables parallel to `ScheduledMessage` that don't have columns like `content` and just have the `sender_id`, `message_id`, `sending_client`, and have this feature be a simplified/parallel implementation to scheduled messages in the server.

OR alternatively, just add a `reminder_target_message_id` optional column on `ScheduledMessage` and have the other fields be values computed when the message was scheduled and exist just for the purpose of displaying previews, but be discarded/replaced with a fresh version when the reminder is actually sent.

Reply:
> OR alternatively, just add a reminder_target_message_id optional column on ScheduledMessage and have the other fields be values computed when the message was scheduled and exist just for the purpose of displaying previews, but be discarded/replaced with a fresh version when the reminder is actually sent.

Done.